### PR TITLE
impl guide: update PVF host page; add diagrams

### PR DESCRIPTION
**Pull request by mrcnski**
Would have been nice to have some helpful diagrams a while ago, but I just thought of it. :P
cc @eagr @Jpserrat 

## Screenshots

<details>

Screenshot 2023-12-01 at 14 46 14

Screenshot 2023-12-01 at 14 46 45

</details>

### CODE REVIEW

**Comment by eagr** [comment]
would be nice if you could also post the clips of the diagrams here :))

**Review by s0me0ne-unkn0wn — APPROVED** [review]
```
   ┌─ node/utility/pvf-prechecker.md:11:110
   │  
11 │   This subsytem does not produce any output messages either. The subsystem will, however, send messages to the [Runtime
   │ ╭──────────────────────────────────────────────────────────────────────────────────────────────────────────────^
12 │ │ API] subsystem to query for the pending PVFs and to submit votes. In addition to that, it will also communicate with
   │ ╰────^ Did you forget to define a URL for `Runtime API`?
   │  
   = hint: declare the link's URL. For example: `[Runtime API]: http://example.com/`
```

Would be good to fix it as well. Otherwise, looks very good!

**Comment by s0me0ne-unkn0wn** [comment]
Probably a random place to ask, but did anybody manage to get a usable PDF of the Implementer's Guide out of `mdbook`? I tried the printing function and `mdbook-pdf` crate, but the diagrams are broken af, they're not auto-scaled, and a lot of them are cut in half.

**Comment by mrcnski** [comment]
@eagr See the "Screenshots section. 😀

@s0me0ne-unkn0wn No idea, but even in HTML mode the diagrams are a bit broken, I need to scroll a lot to see the bigger ones... And it's nice that they are generated from code inline with the docs, but they're a bit ugly haha.

File: polkadot/roadmap/implementers-guide/src/node/utility/pvf-host-and-workers.md [hunk 1]
```diff
@@ -2,12 +2,80 @@
...
+
+```dot process
+digraph {
+ rankdir="LR";
+
+ can [label = "Candidate\nValidation\nSubsystem"; shape = square]
+
+ pvf [label = "PVF Host"; shape = square]
+
+ pp [label = "Prepare\nPool"; shape = square]
```
Comment (eagr) [inline polkadot/roadmap/implementers-guide/src/node/utility/pvf-host-and-workers.md:21]:
```suggestion
pq [label = "Prepare\nQueue"; shape = square]
pvf -> pq [label = "Prepare"; style = dashed]
pq -> pp [label = "Prepare"; style = dashed]
```

**Pull request merged**